### PR TITLE
Make pygame and cv2 dependency optional

### PR DESCRIPTION
**Pull request by Ruishenl**
- Use PIL instead of opencv to decode the encoded bvs to avoid import error when opencv is not installed
- Use Tuple instead of pygame's Vector2 in area_drive to make pygame also an optional dependency. This will pave the way for us to consider packaging the area_drive folder inside invertedai.

I was able to run the ` large_map_example.py` example in a minimum virtualenv with only `invertedai` installed and generating the bv.

@adscib  Let me know what do you think about this.

### CODE REVIEW

File: invertedai/common.py [hunk 1]
```diff
@@ -2,7 +2,9 @@
 from enum import Enum
 from pydantic import BaseModel, model_validator
 import math
-
+from PIL import Image as PImage
+import numpy as np
```
Comment (adscib) [inline invertedai/common.py:6]: According to the `pyproject.toml` file, `numpy` is not a required dependency, so we shouldn't be importing it here at top level. It's a little concerning that the CI passes despite this.

**Review by adscib — CHANGES_REQUESTED** [review]
I generally like the change, as this is a lighter weight approach, but we still need to be careful about which dependencies we require.

File: invertedai/common.py (hunk 1 above)
Comment (Ruishenl) [reply invertedai/common.py:6]: I think it’s because matplotlib is a required dependency and numpy is actually a required package for it.

Comment (adscib) [reply invertedai/common.py:6]: Why is matplotlib required in the first place? Is it just for ScenePlotter? In any case, I think we should make a conscious decision about whether we want to require numpy independently of matplotlib, and if so add it to the dependencies explicitly.

**Comment by Ruishenl** [comment]
I added numpy explicitly as the required package and removed the both matplotlib and numpy from the dev dependencies

**Review by adscib — APPROVED** [review]
I trust that you checked that the birdviews still decode correctly.

**Pull request merged**